main.go

```diff
@@ -11,14 +13,25 @@ import (
 func main() {
 	// this block sets up a go routine to listen for an interrupt signal
 	// which will immediately exit gitleaks
-	stopChan := make(chan os.Signal, 1)
-	signal.Notify(stopChan, os.Interrupt)
-	go listenForInterrupt(stopChan)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Copied from [poutine](https://github.com/boostsecurityio/poutine/blob/cee72a96707d8a389d2c40eee1fa47ea2e63af95/cmd/root.go#L71). I like the "first Ctrl+C gracefully stops, second exits" functionality but it isn't necessary.